### ISSUE TITLE
Test Retrieving from localStorage

### ISSUE THREAD

**Issue by mcshiz** [clarkbw/jest-localstorage-mock#59]
If I am trying to initialize testing my component with some data like so
  
    let obj = {
        date: "2017-12-23T17:33:33.265Z",
        obj: {"hello":"world"}
    };

    localStorage.setItem(obj.date, JSON.stringify(obj.obj));
    expect(localStorage.setItem).toHaveBeenCalledTimes(1); // PASSES

But in my component I am setting the state with what is in localStorage like this:

    let keys = Object.keys(localStorage);
    keys.map((key) => {
            if(Date.parse(key) && moment(key).isValid()) {
                let item = JSON.parse(localStorage.getItem(key));
                item['date'] = moment(key);
                data.push(item)
            }
     });
    this.setState({
            data: data
    })

this.state.data will always be empty because all of the data is set to `localStorage.__STORE__`
How would you go about testing something like this?

**Comment by clarkbw** [comment]
Oh interesting I think this is a bug about the enumerable nature of the localStorage object.

Something like this would remove `__STORE__` from the list of enumerable items.
```js
Object.defineProperty(localStorage, '__STORE__', {
  enumerable: false,
  configurable: false,
  writable: true,
  value: {}
});
```
But then we need to include all the items that have been stored such that you can enumerate through them.

**Comment by mcshiz** [comment]
global.localstorage should only contain key:values which have been set via global.localstorage.setItem(key, val) and not the clear, getItem, setItem properties.
So that way when you iterate through global.localstorage you are only retrieving items set to localstorage

**Comment by mcshiz** [comment]
Something like this
```
class LocalStorage  {
    constructor() {
        Object.defineProperty(this, "getItem", {
            enumerable: false,
            value: function(key)  {
                return this[key] || null
            }
        });
        Object.defineProperty(this, "setItem", {
            enumerable: false,
            value: function(key, val)  {
                this[key] = val + '';
                this.length++
            }
        });
        Object.defineProperty(this, "removeItem", {
            enumerable: false,
            value: function(key) {
                delete this[key];
                this.length++
            }
        });
        Object.defineProperty(this, "clear", {
            enumerable: false,
            value: function() {
                Object.keys(this).map(key => delete this[key]);
                this.length = 0;
            }
        });

        Object.defineProperty(this, "toString", {
            enumerable: false,
            value: function(key) {
                return '[object Storage]'
            }
        });
        Object.defineProperty(this, "key", {
            enumerable: false,
            value: function(idx) {
                return Object.keys(this)[idx] || null
            }
        });
    } // end constructor

}

let local = new LocalStorage();

Object.keys(local)  //[]
local.setItem('hello', 'world')
local.getItem('hello') // 'world'
local.removeItem('hello') //[]
local.setItem('hello', 'world')
Object.keys(local) // ['hello']
local.key(0) // 'hello'
local.clear() //[]
Object.keys(local) //[]
local.toString() // '[object Storage]'
```

P.S. Still trying to figure out how to do length

**Comment by clarkbw** [comment]
Yeah, something like that is where we want to go.  But the class you're working with ☝️ isn't what is returned to clients.

The actual object you're seeing is this one:
https://github.com/clarkbw/jest-localstorage-mock/blob/d774b8f66047cdbebcd37dd6353789803f01b5e5/src/setup.js#L6-L22

Likely just need to translate the class work into that object.

**Comment by clarkbw** [comment]
Fixed by #60
